src/core/document.js

```diff
@@ -765,20 +769,30 @@ class Page {
           );
         }
 
-        const sortedAnnotations = [];
+        let sortedAnnotations = [];
+        const widgetAnnotations = [];
         let popupAnnotations;
         // Ensure that PopupAnnotations are handled last, since they depend on
         // their parent Annotation in the display layer; fixes issue 11362.
         for (const annotation of await Promise.all(annotationPromises)) {
           if (!annotation) {
             continue;
           }
+          if (annotation instanceof WidgetAnnotation) {
+            widgetAnnotations.push(annotation);
+            continue;
+          }
           if (annotation instanceof PopupAnnotation) {
             (popupAnnotations ||= []).push(annotation);
             continue;
           }
           sortedAnnotations.push(annotation);
         }
+        if (sortedAnnotations.length > 0) {
+          sortedAnnotations.push(...widgetAnnotations);
+        } else {
+          sortedAnnotations = widgetAnnotations;
+        }
```

Comment:
```suggestion
        if (widgetAnnotations) {
          sortedAnnotations.push(...widgetAnnotations);
        }
```

Reply:
My idea here was just to avoid a useless push in a potentially empty array (which should be the case with most of the forms).